py_mmd_tools/odajson_to_mmd.py

```diff
@@ -0,0 +1,306 @@
+            'file '
+            'aborted.')
+        return None
+
+    # Merge default and dataset specific elements
+    # (default is overwritten if it's also defined at dataset level)
+    elements_out = copy.deepcopy(default_elements)
+    _merge_dicts(elements_out, dataset_elements)
+
+    # Rename elements to follow mmd vocabulary for data production status
+    choices = {'open': 'In Work', 'future': 'Planned', 'closed': 'Complete'}
+    elements_out['dataset_production_status'] = choices.get(elements_out['production_status'], None)
+    if elements_out['dataset_production_status'] is None:
+        logger.warning(
+            'The value for key %s (%s) is not a valid choice. Correct choices are: '
+            'open/future/closed. This element '
+            'is required, so the creation of MMD file is aborted.' % (
+                'production_status', elements_out['production_status']))
+        return None
+
+    # Remove operational status if empty
+    if elements_out['operational_status'] == '':
+        elements_out.pop('operational_status')
+    else:
+        elements_out['operational_status'] = elements_out['operational_status'].title()
+
+    # Remove end date for ongoing data
+    elements_out['temporal_extent'].pop('end_date', None)
+
+    # Create dataset specific title and abstract if possible
+    try:
+        elements_out['title_full'] = elements_out[
+                                         'keywords_cf'] + ' observations from weather station ' + \
+                                     elements_out['station_name'] + ' (station ID ' + elements_out[
+                                         'station_id'] + ').'
+        elements_out['abstract_full'] = 'Timeseries of ' + elements_out[
+            'keywords_cf'] + ' observations from the Norwegian weather station ' + elements_out[
+                                            'station_name'] + ' (station ID ' + elements_out[
+                                            'station_id'] + '). The observations have been ' \
+                                                            'through the data ' \
+                                                            'collection system of the Norwegian ' \
+                                                            'Meteorological ' \
+                                                            'institute which includes a number of ' \
+                                                            'automated and ' \
+                                                            'manual quality control routines. The ' \
+                                                            'number of available ' \
+                                                            'quality control routines is element ' \
+                                                            'dependent. '
+    except KeyError:
+        elements_out['title_full'] = elements_out['title']
```

Comment:
would it be better to switch the order? In case obsklim changes the title and/or abstract, that would then have preference. Also, it is good style to only have one command for each try clause.
```suggestion
try:
    elements_out['title_full'] = elements_out['title']
except KeyError:
            elements_out['title_full'] = elements_out[
                       'keywords_cf'] + ' observations from weather station ' + \
                       elements_out['station_name'] + ' (station ID ' + elements_out[
                       'station_id'] + ').'
```

Reply:
The plan was to have a title and abstract specific to the dataset (ie with name of elements, id, ...) so the order is right. Actually, maybe I should get rid of the default title and abstract as they are not very meaningful.